test/filterparser.cpp

```diff
@@ -64,3 +64,79 @@ TEST_CASE("Both = and == are accepted", "[FilterParser]")
 	REQUIRE(fp.parse_string("a = \"abc\""));
 	REQUIRE(fp.parse_string("a == \"abc\""));
 }
+
+TEST_CASE("FilterParser disallows NUL byte inside filter expressions",
+	"[FilterParser]")
+{
+	FilterParser fp;
+
+	REQUIRE_FALSE(fp.parse_string(std::string("attri\0bute = 0", 14)));
+	REQUIRE_FALSE(fp.parse_string(std::string("attribute\0= 0", 13)));
+	REQUIRE_FALSE(fp.parse_string(std::string("attribute = \0", 13)));
+	REQUIRE_FALSE(fp.parse_string(std::string("attribute = \\\"\0\\\"", 17)));
+
+	// The following shouldn't pass, but it does. Further REQUIREs explain why:
+	// the NUL byte silently terminates parsing.
+	REQUIRE(fp.parse_string(std::string("attribute = \"hello\0world\"", 25)));
```

Comment:
Is this something you want to get fixed?

Reply:
Yes. It will be automatically fixed once we moved enough code to Rust: when config line is read straight into Rust `String` and then passed around without conversions to `char*`, internal NUL bytes won't be a problem anymore.

We could expedite this by passing strings' lengths over FFI, but I doubt it's worth the trouble. Never had a bug report from a user who wanted to use NUL in their filter expressions.

Reply:
> Yes. It will be automatically fixed once we moved enough code to Rust

Sounds good